### ISSUE TITLE
Review comment tools for .Rmd

### ISSUE THREAD

**Issue by MilesMcBain** [ropensci/unconf17#86]
Something that is really holding back the more cautious folk from jumping into the .Rmd workflow for papers is the lack of user friendly comment and review tools. There is currently no easy way to:

* Mark up .Rmd output with tagged comments that identify the commenter, time etc.
* Map comments to specific code/text
* Highlight new additions/deletions since the previous review.

The comment thing in particular gets mentioned on Twitter periodically. I've had a bash at something lofi [here](https://github.com/MilesMcBain/RmdComment).

@njtierney recently introduced me to another Oz researcher who has created a Shiny app that hosts compiled .Rmd output and captures comments in text inputs. Again it's pretty lofi but it tells you something about the perceived value of this feature that they were prepared to go that far.

Please hit me with any other work in this space you know about.

I imagine most of us here are fine with git/Github for this purpose, but we have to remember how deep into the tail of reproducible research we really are.

**Comment by stephlocke** [comment]
Being able to add comments would be a nice thing - I miss it from working with latex http://ctan.sharelatex.com/tex-archive/macros/latex/contrib/todonotes/todonotes.pdf

**Comment by MilesMcBain** [comment]
Yes! I was also directed to [latexdiff](https://www.sharelatex.com/blog/2013/02/16/using-latexdiff-for-marking-changes-to-tex-documents.html) recently which seems like .Rmd should be able to do.

**Comment by hadley** [comment]
A somewhat related problem is round-tripping Rmds to google docs.

**Comment by markdly** [comment]
If you slighty tweak the setup of a regular .Rmd file it's possible to embed commenting functionality in generated html output. 

I've put up some examples to try [here](https://markdly.github.io/reviewr/index.html) which use the ready made javascript libraries [annotatorjs](http://annotatorjs.org/) and the [hypothesis client](https://github.com/hypothesis/client).

At this stage the examples only allow adding comments to the .Rmd output.  But perhaps it's possible to extend this so the comments can be mapped back to the .Rmd source file somehow ...
